apps/web/middleware.ts

```diff
@@ -3,13 +3,31 @@ import { collectEvents } from "next-collect/server";
 import type { NextMiddleware } from "next/server";
 import { NextResponse, userAgent } from "next/server";
 
+import { orgDomainConfig } from "@calcom/features/ee/organizations/lib/orgDomains";
 import { CONSOLE_URL, WEBAPP_URL, WEBSITE_URL } from "@calcom/lib/constants";
 import { isIpInBanlist } from "@calcom/lib/getIP";
 import { extendEventData, nextCollectBasicSettings } from "@calcom/lib/telemetry";
 
 const middleware: NextMiddleware = async (req) => {
   const url = req.nextUrl;
   const requestHeaders = new Headers(req.headers);
+  const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(req.headers.get("host") ?? "");
+
+  // Make sure we are in the presence of an organization
+  if (isValidOrgDomain) {
+    // In the presence of an organization, cover its profile page at "/"
+    if (url.pathname === "/") {
+      // rewrites for org profile page using team profile page
+      if (
+        url.pathname.indexOf("/api") === -1 &&
+        url.pathname.indexOf("/auth") === -1 &&
+        url.pathname.indexOf("/apps") === -1
+      ) {
+        url.pathname = `/org/${currentOrgDomain}`;
+        return NextResponse.rewrite(url);
+      }
+    }
+  }
```

Comment:
This can simplified to
```suggestion
 // In the presence of an organization, cover its profile page at "/"
  if (isValidOrgDomain && url.pathname === "/") {
      // rewrites for org profile page using team profile page
      url.pathname = `/org/${currentOrgDomain}`;
      return NextResponse.rewrite(url);
  }
```
this condition will always be true 
`if (
        url.pathname.indexOf("/api") === -1 &&
        url.pathname.indexOf("/auth") === -1 &&
        url.pathname.indexOf("/apps") === -1
      )`

Reply:
Done, thanks Efrain!